src/components/pages/HomePage/TeachersButton.tsx

```diff
@@ -1,27 +1,20 @@
 /** @jsxImportSource @emotion/react */
 
-import { Button, Typography } from '@mui/material';
+import { Button } from '@mui/material';
 import { School as SchoolIcon } from '@mui/icons-material';
-import { useState, useRef } from 'react';
 
-import { getClassroom } from '@utils/classrooms';
-import BasicModal from '@components/shared/Modal';
-import ModalTextField from '@components/shared/ModalTextField';
+function getRandomPin(pinLength: number) {
+  const maxWholeNumber = Math.pow(10, pinLength);
+  const randomDecimal = Math.random();
+  const randomPin = Math.floor(randomDecimal * maxWholeNumber);
+  const randomPinStr = randomPin.toString();
+  const zeroPaddedPin = randomPinStr.padStart(pinLength, '0');
+  return zeroPaddedPin;
+}
```

Comment:
This is more readable for me than the initial version. Although I'd forgotten what Math.power and padStart did and had to look them up. The version below uses more common Javascript methods.  That being said, I defer to you and you can use whichever version you think is best!

function getRandomPin(pinLength: number) {
  const randomDigit = () => Math.floor(Math.random() * 10);
  const pin = [];
  for (let i = 0; i < pinLength; i++)
    pin.push(randomDigit());
  return pin.join('');
}

Reply:
If the issue is readability, a for loop will never be more readable than a flat list of verbose steps.

Additionally, the variable names tell the story; you don't need to know how the functions like Math.pow() and padStart work in order to be able to read the code.  It can easily be seen what is intended.

How does your loop version accomplish this level of readability? 

If the tables were turned and I was reviewing your loop version of getRandomPin I would definitely be pushing you to rewrite it without using a for loop as it is unnecessary for many reasons and does not improve readability.

Calling Math.random wrapped in Math.floor repeatedly inside a for loop is not performant.  I know it doesn't matter here, but I would never do this ever, it's just bad practice.

However, in the spirit of what seems to be a readability issue, I've refactored this function again.

I think my latest version is the simplest yet.

```typescript
function getRandomPin(pinLength: number) {
  const randomPin = Math.random().toString();
  return randomPin.slice(2, pinLength + 2);
}
```

Reply:
Thanks, I learned a lot by reading your explanation. Looking back at it, I see what you mean that the variable names tell the story. Also, I see what you mean that calling the extra methods repeatedly in the for loop is not performant. Hadn't thought of that but now I see it.  Thanks!

Agreed that the latest version is the simplest yet! By far and away the winner.